app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharacterView.kt

```diff
@@ -5,15 +5,18 @@ import android.view.View
 import com.puzzlebench.clean_marvel_kotlin.R
 import com.puzzlebench.cmk.domain.model.Character
 import com.puzzlebench.clean_marvel_kotlin.presentation.MainActivity
+import com.puzzlebench.clean_marvel_kotlin.presentation.TAG
 import com.puzzlebench.clean_marvel_kotlin.presentation.adapter.CharacterAdapter
 import com.puzzlebench.clean_marvel_kotlin.presentation.extension.showToast
-import kotlinx.android.synthetic.main.activity_main.*
+import com.puzzlebench.clean_marvel_kotlin.presentation.fragments.CharacterFragmentDialog
+import kotlinx.android.synthetic.main.activity_main.progressBar
+import kotlinx.android.synthetic.main.activity_main.recycleView
 import java.lang.ref.WeakReference
 
 class CharacterView(activity: MainActivity) {
     private val activityRef = WeakReference(activity)
     private val SPAN_COUNT = 1
-    var adapter = CharacterAdapter { character -> activity.applicationContext.showToast(character.name) }
+    var adapter = CharacterAdapter { character -> showFragmentDialog(character) }
 
     fun init() {
```

Comment:
you can use the kotlin style here and make something like
activityRef.get()?.let {
     it.recycleView.layoutManager = GridLayoutManager(activity, SPAN_COUNT)
     it.recycleView.adapter = adapter
     showLoading()
}

Reply:
done